### ISSUE TITLE
Question: [jsdoc/require-jsdoc] Debug / find correct context to apply

### ISSUE THREAD

**Issue by Shinigami92** [gajus/eslint-plugin-jsdoc#751]
I have following code:

```ts
import { defineComponent, PropType } from '@vue/composition-api';

export default defineComponent({
  setup() {
    return {};
  }
});
```

![image](https://user-images.githubusercontent.com/7195563/120996872-04cd2e00-c787-11eb-834c-6e9c6ccbf404.png)

My config:
```ts
'jsdoc/require-jsdoc': [
  'warn',
  {
    contexts: [
      'ClassDeclaration',
      'ClassProperty:not([accessibility=/(private|protected)/])',
      'ExportNamedDeclaration:has(VariableDeclaration)',
      ':not([name="setup"]) ~ FunctionExpression', // <---
      'MethodDefinition:not([accessibility=/(private|protected)/]) > FunctionExpression',
      'TSEnumDeclaration',
      'TSInterfaceDeclaration',
      'TSMethodSignature',
      // 'TSPropertySignature',
      'TSTypeAliasDeclaration'
    ]
  }
]
```

I want to disable `jsdoc/require-jsdoc` if `FunctionExpression` has pre-sibling with `name !== setup`
Sadly it just turns off also for non `setup` 🙁 and it's kinda the same as just remove `'FunctionExpression'` from contexts

Do you know how I can configure that?

This is needed cause I want to write JSDoc for functions that are written by us, but not for e.g. `setup` that is predefined by Vue and just need to be defined

**Comment by brettz9** [comment]
The sibling operator only comes into play if the AST is within an array.

If you are looking for this on `Property`, you can indicate:

```js
  {
    contexts: ['Property[key.name!="setup"][value.type="FunctionExpression"]'],
  }
```

Closing, as I think that should resolve, but feel free to comment further.

**Comment by Shinigami92** [comment]
It works 😃
